### PR TITLE
feat: 🎸 generate info if dataset_infos.json does not exist

### PR DESCRIPTION
**Pull request by severo**
in get_dataset_infos(). Also: add the `use_auth_token` parameter, and create get_dataset_config_info()

✅ Closes: #3013

### CODE REVIEW

**Comment by severo** [comment]
It's a first attempt at solving https://github.com/huggingface/datasets/issues/3013.

File: src/datasets/inspect.py [hunk 1]
```diff
@@ -154,21 +156,36 @@ def get_dataset_infos(
...
+            revision=revision,
+            download_config=download_config,
+            download_mode=download_mode,
+            data_files=data_files,
+            config_name=config_name,
+            data_dir=None,
+            cache_dir=None,
+            features=None,
+            use_auth_token=use_auth_token,
+            # **download_kwargs,
```
Comment (lhoestq) [inline src/datasets/inspect.py:185]: You can remove this line

**Review by lhoestq — COMMENTED** [review]
Cool thank you ! Do you mind adding some tests for these functions ?

File: src/datasets/inspect.py (hunk 1 above)
Comment (lhoestq) [reply src/datasets/inspect.py:185]: I think we need at least those ones: path/revision/config_name (obviously), download_config/download_mode_use_auth_token (it downloads the dataset script from the hub), data_files/data_dir (splits may depend on these)

So maybe these two could be potentially removed:
- cache_dir: where to download data files and arrow files (not necessary when we just check the splits or the configs)
- features: overwrite the default feature type of the dataset (completely useless in this case)

what do you think ?

Comment (severo) [reply src/datasets/inspect.py:185]: data_dir: in this function (get_dataset_infos) it does not make sense either, right, since it's related to a config?

Comment (severo) [reply src/datasets/inspect.py:185]: My point the mismatch between the two function signatures:

```
def get_dataset_infos(
    path: str,
    revision: Optional[Union[str, Version]] = None,
    download_config: Optional[DownloadConfig] = None,
    download_mode: Optional[GenerateMode] = None,
    force_local_path: Optional[str] = None,
    dynamic_modules_path: Optional[str] = None,
    data_files: Optional[Union[Dict, List, str]] = None,
    use_auth_token: Optional[Union[bool, str]] = None,
    **download_kwargs,
):
```

and 

```
def get_dataset_config_info(
    path: str,
    config_name: Optional[str] = None,
    data_dir: Optional[str] = None,
    data_files: Optional[Union[str, Sequence[str], Mapping[str, Union[str, Sequence[str]]]]] = None,
    cache_dir: Optional[str] = None,
    features: Optional[Features] = None,
    download_config: Optional[DownloadConfig] = None,
    download_mode: Optional[GenerateMode] = None,
    revision: Optional[Union[str, Version]] = None,
    use_auth_token: Optional[Union[bool, str]] = None,
    **config_kwargs,
) -> DatasetInfo:
```

The first one calls the second one

Comment (lhoestq) [reply src/datasets/inspect.py:185]: I'll check what parameters we need and update the PR, will ping you once it's done :)
Thanks for raising this concern

Comment (severo) [reply src/datasets/inspect.py:185]: thank you!

**Comment by lhoestq** [comment]
I only kept these ones:
```
    path: str,
    data_files: Optional[Union[Dict, List, str]] = None,
    download_config: Optional[DownloadConfig] = None,
    download_mode: Optional[GenerateMode] = None,
    revision: Optional[Union[str, Version]] = None,
    use_auth_token: Optional[Union[bool, str]] = None,
    **config_kwargs,
```

Let me know if it's better for you now !

(note that there's no breaking change since the ones that are removed can be passed as config_kwargs if you really want)

**Comment by severo** [comment]
(https://github.com/huggingface/datasets/pull/3670/commits/07c3f0800dd34dfebb9674ad46c67a907b08ded8 -> I has forgotten to update black in my venv)

**Pull request merged**